### PR TITLE
♻️ 🐛 [RUMF-1178] improve logs assembly - Part 1

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

First step of the logs assembly refactoring
[RUMF-576] Fix the logs printed twice in the console when logs 'forwardErrorsToLogs' is enabled

## Changes

- Extract log assembly from startLogs
- Extract the send/print logic from the logger with a Sender delegate (needed for the fix)

## Testing

- [x] Local
- [ ] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

[RUMF-576]: https://datadoghq.atlassian.net/browse/RUMF-576?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/logs/src/domain/sender.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  }
+  const contextManager = createContextManager()
+  contextManager.set(loggerContext)
+
+  function isAuthorized(status: StatusType, handlerType: HandlerType) {
+    const sanitizedHandlerType = Array.isArray(conf.handlerType) ? conf.handlerType : [conf.handlerType]
+    return STATUS_PRIORITIES[status] >= STATUS_PRIORITIES[conf.level] && includes(sanitizedHandlerType, handlerType)
+  }
+
+  function sendHttpRequest(message: string, messageContext: object | undefined, status: StatusType) {
```
Comment (BenoitZugmeyer) [inline packages/logs/src/domain/sender.ts:33]: **💬 suggestion:** ‏Naming: maybe we could have `sendToHttp` / `sendToConsole`?

File: packages/logs/src/domain/sender.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+export const STATUS_PRIORITIES: { [key in StatusType]: number } = {
+  [StatusType.debug]: 0,
+  [StatusType.info]: 1,
+  [StatusType.warn]: 2,
+  [StatusType.error]: 3,
+}
+
+export function buildSender(
+  sendLog: (message: LogsMessage) => void,
```
Comment (BenoitZugmeyer) [inline packages/logs/src/domain/sender.ts:16]: **💬 suggestion:** Naming: ‏`sendLog` is also used as a returned property, with a different meaning. Maybe we can find a more explicit name? Like `sendToHttpImpl`?

File: packages/logs/src/boot/startLogs.ts [hunk 3]
```diff
@@ -164,7 +155,7 @@ export function doStartLogs(
         },
       }
     }
-    logger.log(log.message, messageContext, LogStatusForApi[log.api])
+    sender.sendHttpRequest(log.message, messageContext, LogStatusForApi[log.api])
```
Comment (BenoitZugmeyer) [inline packages/logs/src/boot/startLogs.ts:158]: **👏 praise:** ‏I like that :)

File: packages/logs/src/domain/sender.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+export type Sender = ReturnType<typeof buildSender>
+
+export const STATUS_PRIORITIES: { [key in StatusType]: number } = {
+  [StatusType.debug]: 0,
+  [StatusType.info]: 1,
+  [StatusType.warn]: 2,
+  [StatusType.error]: 3,
+}
+
+export function buildSender(
```
Comment (BenoitZugmeyer) [inline packages/logs/src/domain/sender.ts:15]: **💬 suggestion:** What about naming it `createSender`? I feel like `build` is more about formating immutable values, where `create` is to define a live value with some mutable state. WDYT?

File: packages/logs/src/domain/sender.ts (hunk 1 above)
Comment (amortemousque) [reply packages/logs/src/domain/sender.ts:33]: Yes better

File: packages/logs/src/domain/sender.ts (hunk 2 above)
Comment (amortemousque) [reply packages/logs/src/domain/sender.ts:16]: Sure

File: packages/logs/src/domain/sender.ts (hunk 4 above)
Comment (amortemousque) [reply packages/logs/src/domain/sender.ts:15]: Let's do that ;)

**Review by BenoitZugmeyer — APPROVED** [review]

**Pull request merged**